tests/rule_based_profiler/parameter_builder/test_regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,249 @@
+from typing import Set
+
+import pytest
+
+import great_expectations.exceptions.exceptions as ge_exceptions
+from great_expectations.data_context import DataContext
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.parameter_builder import (
+    RegexPatternStringParameterBuilder,
+    regex_pattern_string_parameter_builder,
+)
+from great_expectations.rule_based_profiler.types import (
+    Domain,
+    ParameterContainer,
+    get_parameter_value_by_fully_qualified_parameter_name,
+)
+
+
+def test_regex_pattern_string_parameter_builder_instantiation_with_defaults():
+    candidate_regexes: Set[str] = {
+        r"/\d+/",  # whole number with 1 or more digits ExpectValuesToBeNumeric? (.. you would want to emit that expectation)?
+        r"/-?\d+/",  # negative whole numbers
+        r"/-?\d+(\.\d*)?/",  # decimal numbers with . (period) separator
+        r"/[A-Za-z0-9\.,;:!?()\"'%\-]+/",  # general text
+        r"^ +/",  # leading space
+        r" +/$",  # trailing space
+        r"/https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#()?&//=]*)/",  # Matching URL (including http(s) protocol)
+        r"/<\/?(?:p|a|b|img)(?: \/)?>/",  # HTML tags
+        r"/(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})(?:.(?:25[0-5]|2[0-4]\d|[01]\d{2}|\d{1,2})){3}/",  # IPv4 IP address
+        r"/(?:[A-Fa-f0-9]){0,4}(?: ?:? ?(?:[A-Fa-f0-9]){0,4}){0,7}/",  # IPv6 IP address,
+    }
+    regex_pattern_string_parameter: RegexPatternStringParameterBuilder = (
+        RegexPatternStringParameterBuilder(
+            name="my_simple_regex_string_parameter_builder",
+            candidate_regexes=candidate_regexes,
+        )
+    )
+
+    assert regex_pattern_string_parameter._threshold == 1.0
+    assert regex_pattern_string_parameter._candidate_regexes == candidate_regexes
```

Comment:
@Shinnnyshinshin Please update in accordance to now being able to access properties (public methods).  Assertions might need to change, because the logic/style will change.  Thanks.

Reply:
Done :)